sdk/core/core-paging/src/getPagedAsyncIterator.ts

```diff
@@ -28,26 +30,38 @@ export function getPagedAsyncIterator<
     },
     byPage:
       pagedResult?.byPage ??
-      ((settings?: PageSettings) => {
+      (((settings?: PageSettings) => {
         const { continuationToken, maxPageSize } = settings ?? {};
-        return getPageAsyncIterator(pagedResult as PagedResult<TPage, PageSettings, TLink>, {
-          pageLink: continuationToken as unknown as TLink | undefined,
-          maxPageSize,
-        });
-      }),
+        return getPageAsyncIterator(
+          pagedResult as PagedResult<TPage, PageSettings, TLink, TElement>,
+          {
+            pageLink: continuationToken as unknown as TLink | undefined,
+            maxPageSize,
+          }
+        );
+      }) as unknown as (settings?: TPageSettings) => AsyncIterableIterator<TPage>),
   };
 }
 
 async function* getItemAsyncIterator<TElement, TPage, TLink, TPageSettings>(
-  pagedResult: PagedResult<TPage, TPageSettings, TLink>
+  pagedResult: PagedResult<TPage, TPageSettings, TLink, TElement>
 ): AsyncIterableIterator<TElement> {
   const pages = getPageAsyncIterator(pagedResult);
   const firstVal = await pages.next();
   // if the result does not have an array shape, i.e. TPage = TElement, then we return it as is
   if (!Array.isArray(firstVal.value)) {
-    yield firstVal.value;
-    // `pages` is of type `AsyncIterableIterator<TPage>` but TPage = TElement in this case
-    yield* pages as unknown as AsyncIterableIterator<TElement>;
+    // can extract elements from this page
+    const toElements = pagedResult.toElements;
```

Comment:
nit `const { toElements } = pagedResult;` more modern javascript-ish?

Reply:
yup this is a more concise representation. The feature is called [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).